src/DataWrangling/ecco_metadata.jl

```diff
@@ -0,0 +1,173 @@
+    elseif metadata.version isa ECCO2Daily
+        daystr = string(Dates.day(metadata.dates), pad=2)
+        return shortname * postfix * yearstr * monthstr * daystr * ".nc"
+    end
+end
+
+# Convenience functions
+short_name(data::ECCOMetadata{<:Any, <:ECCO2Daily})   = ecco2_short_names[data.name]
+short_name(data::ECCOMetadata{<:Any, <:ECCO2Monthly}) = ecco2_short_names[data.name]
+short_name(data::ECCOMetadata{<:Any, <:ECCO4Monthly}) = ecco4_short_names[data.name]
+
+field_location(data::ECCOMetadata) = ecco_location[data.name]
+
+variable_is_three_dimensional(data::ECCOMetadata) = 
+    data.name == :temperature || 
+    data.name == :salinity || 
+    data.name == :u_velocity ||
+    data.name == :v_velocity
+
+ecco4_short_names = Dict(
+    :temperature           => "THETA",
+    :salinity              => "SALT",
+    :u_velocity            => "EVEL",
+    :v_velocity            => "NVEL",
+    :sea_ice_thickness     => "SIheff",
+    :sea_ice_area_fraction => "SIarea"
+)
+
+ecco2_short_names = Dict(
+    :temperature           => "THETA",
+    :salinity              => "SALT",
+    :u_velocity            => "UVEL",
+    :v_velocity            => "VVEL",
+    :sea_ice_thickness     => "SIheff",
+    :sea_ice_area_fraction => "SIarea"
+)
+
+ecco_location = Dict(
+    :temperature           => (Center, Center, Center),
+    :salinity              => (Center, Center, Center),
+    :sea_ice_thickness     => (Center, Center, Nothing),
+    :sea_ice_area_fraction => (Center, Center, Nothing),
+    :u_velocity            => (Face,   Center, Center),
+    :v_velocity            => (Center, Face,   Center),
+)
+
+# URLs for the ECCO datasets specific to each version
+urls(::ECCOMetadata{<:Any, <:ECCO2Monthly}) = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/monthly/"
+urls(::ECCOMetadata{<:Any, <:ECCO2Daily})   = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/daily/"
+urls(::ECCOMetadata{<:Any, <:ECCO4Monthly}) = "https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/interp_monthly/"
+
+"""
+    download_dataset!(metadata::ECCOMetadata)
+
+Download the dataset specified by the given metadata. If the metadata contains a single date, 
+the dataset is downloaded directly. If the metadata contains multiple dates, the dataset is 
+downloaded for each date individually. 
+The data download requires a username and password to be provided in the ECCO_USERNAME and ECCO_PASSWORD
+environment variables. This can be done by exporting the environment variables in the shell before running the script,
+or by launching julia with 
+
+ECCO_USERNAME=myuser ECCO_PASSWORD=mypasswrd julia 
+
+# Arguments
+- `metadata::ECCOMetadata`: The metadata specifying the dataset to be downloaded.
+"""
+function download_dataset!(metadata::ECCOMetadata;
+                           username = get(ENV, "ECCO_USERNAME", nothing), 
+                           password = get(ENV, "ECCO_PASSWORD", nothing),
```

Comment:
is this insecure? This invites people to write their passwords in a script which could get uploaded to git.

You may want to remove the option of providing `username` and `password` via kwarg.

Reply:
removed!